src/components/CheckEditor/transformations/toFormValues.grpc.ts

```diff
@@ -14,10 +17,12 @@ export function getGRPCCheckFormValues(check: GRPCCheck): CheckFormValuesGRPC {
 }
 
 const getGRPCSettingsFormValues = (settings: GRPCCheck['settings']): GRPCSettingsFormValues => {
+  const transformedTlsConfig = getTlsConfigFormValues(settings.grpc?.tlsConfig);
+
   return {
     ipVersion: settings.grpc?.ipVersion,
     service: settings.grpc?.service,
     tls: settings.grpc?.tls,
-    tlsConfig: settings.grpc?.tlsConfig,
+    ...transformedTlsConfig,
```

Comment:
And here 😅 (this also needs some tests to catch it as mentioned in the future work part of the PR).